### PR TITLE
SDK-2374 added failure reason info to idv and added tests

### PR DESCRIPTION
**Pull request by mehmet-yoti**

### CODE REVIEW

File: src/Yoti.Auth/DocScan/Session/Retrieve/IdentityProfile/RequirementNotMetDetails.cs
Comment (saurabh-yoti) [inline src/Yoti.Auth/DocScan/Session/Retrieve/IdentityProfile/RequirementNotMetDetails.cs:1]: Properties are already public and have getters. So, separate Get methods like `GetFailureType()` and `GetDetails()` are not required.

File: test/Yoti.Auth.Tests/DocScan/Session/Retrieve/IdentityProfile/FailureReasonResponseTest.cs
Comment (saurabh-yoti) [inline test/Yoti.Auth.Tests/DocScan/Session/Retrieve/IdentityProfile/FailureReasonResponseTest.cs:1]: Please update the test after removing redundant Get methods in `RequirementNotMetDetails.cs`.

**Review by saurabh-yoti — CHANGES_REQUESTED** [review]

File: src/Yoti.Auth/DocScan/Session/Retrieve/IdentityProfile/RequirementNotMetDetails.cs
Comment (mehmet-yoti) [reply src/Yoti.Auth/DocScan/Session/Retrieve/IdentityProfile/RequirementNotMetDetails.cs:1]: solved

File: test/Yoti.Auth.Tests/DocScan/Session/Retrieve/IdentityProfile/FailureReasonResponseTest.cs
Comment (mehmet-yoti) [reply test/Yoti.Auth.Tests/DocScan/Session/Retrieve/IdentityProfile/FailureReasonResponseTest.cs:1]: solved

**Review by saurabh-yoti — APPROVED** [review]

**Pull request merged**